app/models/piece.rb

```diff
@@ -27,6 +27,22 @@ def off_board?(new_x, new_y)
     new_x < 1 || new_x > 8 || new_y < 1 || new_y > 8
   end
 
+  def move_to!(new_x, new_y)
+    @moving_piece = self.piece
+    @moving_piece_x_position = self.x_position
+    @moving_piece_y_position = self.y_position
+    if (new_x != nil) && (new_y != nil)
+      @piece_at_destination = @game.pieces.find_by(x_position: new_x, y_position: new_y)
+      if self.color === @piece_at_destination.color
+        return "Error"
+      else
+        @piece_at_destination.update_attributes(x_position: nil, y_position: nil)
```

Comment:
`def capture_piece(captured_piece)`
      `captured_piece.update(x_position:  nil, y_position: nil)`
 `end`

Reply:
Good suggestion @PaleoGuy2012 - such a method `capture_piece` (maybe even with a ! at the end) makes it much more clear what happens

Reply:
Didn't think to just make a new method, but that's way better, thanks!